iterators_task_2/student/include/impl.h

```diff
@@ -4,12 +4,38 @@
 template <class Container>
 struct accumulator
 {
+protected:
+    Container& m_container;
+
+public:
     /** @todo Iterator traits*/
+    typedef typename Container::value_type value_type;
```

Comment:
1. Prefer using
2. Usually traits are void (only tag is provided) for output iterators